syncer/syncer.go

```diff
@@ -979,6 +1006,11 @@ func (s *Syncer) addJob(job *job) error {
 		startTime := time.Now()
 		s.tctx.L().Debug("queue for key", zap.Int("queue", queueBucket), zap.String("key", job.key))
 		s.jobs[queueBucket] <- job
+		s.isTransactionEnd.Store(false)
```

Comment:
Will this scene cause a problem?
1. run `s.jobs[queueBucket] <- job` first, dml are sent to `jobs` channel.
2. `jobs` goroutine starts to execute this dml SQL. Some sqls are written to downstream.
3. `s.isTransactionEnd.Load()` is true, `runCancel()` directly.
4. run L1009, `s.isTransactionEnd.Store(false)`

Some sqls out of this transaction are still written to downstream.

Reply:
You can see the latest commit https://github.com/pingcap/dm/pull/1928/commits/d0f7737dfa6874a4a8aaa7ff5f71a3ae81281513 whether solve this problem.

Reply:
Maybe:
```
                                  <- xid event
                                  <- s.isTransactionEnd.Store(true)
if s.isTransactionEnd.Load() {    <- enter if
	                              <- dml event
                                  <- s.isTransactionEnd.Store(false)
	                              <- s.jobs[queueBucket] <- job
	s.jobWg.Wait()
	tctx.L().Info("the last job is transaction end, done directly")
	runCancel()
	s.waitTransactionLock.Unlock()
	return
}
s.isTransactionEnd.Load()
```

also will write other sqls to downstream.

I also thought about that 

```
if s.isTransactionEnd.Load() {
    s.waitXIDJob.Store(int64(waitComplete)) // add
	s.jobWg.Wait()
	tctx.L().Info("the last job is transaction end, done directly")
	runCancel()
	s.waitTransactionLock.Unlock()
	return
}
```

However, once the `s.waitXIDJob.Store(int64(waitComplete))` executed after the check https://github.com/pingcap/dm/pull/1928/files#diff-e725a03bd475dc4ea28028613ede1fd88a8cc8f881f0a1788a189e766f191830R1753, the problem also exits.